File: section-04-basic-arrays/01-sum-of-array-elements/sum-of-array-elements.py
class Solution:
    # Function to get the sum of array elements
    def sum(self, arr, n):
        total = 0  # to store the answer

        # Iterate on all the elements
        for i in range(n):
            # Add the current element to the sum
            total = total + arr[i]

        # Return the result
        return total


# Main method
if __name__ == "__main__":
    n = 4
    arr = [1, 2, 3, 8]

    # Creating an instance of solution class
    sol = Solution()

    # Function call to get the sum of array elements
    result = sol.sum(arr, n)

    # output the result
    print("The sum of array elements is:", result)

File: section-04-basic-arrays/01-sum-of-array-elements/sum-of-array-elements.cs
using System;

public class Solution
{
    // Function to get the sum of array elements
    public int sum(int[] arr, int n)
    {
        int ans = 0; // to store the answer

        // Iterate on all the elements
        for (int i = 0; i < n; i++)
        {
            // Add the current element to the sum
            ans = ans + arr[i];
        }

        // Return the result
        return ans;
    }
}




/* 

Complexity Analysis
Time Complexity : O(N), because each element in the array is processed exactly once.
Space Complexity : O(1), because only couple of variable are used.

 */